pennylane/transforms/control.py

```diff
@@ -0,0 +1,191 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the control transform.
+"""
+
+
+from functools import wraps
+from pennylane.tape import QuantumTape
+from pennylane.operation import Operation, AnyWires
+from pennylane.wires import Wires
+from pennylane.transforms.registrations import register_control, CONTROL_MAPS
+from pennylane.transforms.adjoint import adjoint
+
+
+def expand_with_control(tape, control_wire):
+    """Expand a tape to include a control wire on all queued operations.
+
+    Args:
+        tape: A QuantumTape.
+        control_wire: A single wire to use as the control wire.
+    Returns:
+        A new QuantumTape with the controlled operations.
+    """
+    with QuantumTape(do_queue=False) as new_tape:
+        for op in tape.operations:
+            if op.__class__ in CONTROL_MAPS:
+                # Create the controlled version of the operation
+                # and add that the to the tape context.
+                # NOTE(chase): Should we use type(op) here instead?
+                CONTROL_MAPS[op.__class__](op, control_wire)
+            else:
+                raise NotImplementedError(
+                    f"Control transform of operation {type(op)} is not registered."
+                    "You can define the control transform by using qml.register_control"

```

Comment:
Note: `register_control` is not top level, so this will not work. We have two options:

* Import `register_control`
* Make `qml.transforms` a visible subpackage for users, and add a module page to the docs

Reply:
Both sound good to me.